src/lib.rs

```diff
@@ -463,6 +468,122 @@ impl<'a, T: 'a> NodeMut<'a, T> {
         unsafe { self.tree.get_unchecked(self.id).has_children() }
     }
 
+    /// Apply function for each ancestor mutable node reference.
+    pub fn for_each_ancestor<F>(&mut self, mut f: F)
+    where
+        F: FnMut(&mut NodeMut<T>),
+    {
+        if let Some(mut parent_node) = self.parent() {
+            f(&mut parent_node);
+            while let Ok(mut n) = parent_node.into_parent() {
+                f(&mut n);
+                parent_node = n;
+            }
+        }
```

Comment:
```suggestion
        let current = self;
        while let Ok(mut parent) = current.into_parent() {
            f(&mut parent);
            current = parrent;
        }
```

to avoid the loop-and-a-half pattern.

Reply:
(I think this applies to the other functions in the same way.)